packages/core/content-releases/admin/src/pages/ReleasesPage.tsx

```diff
@@ -1,25 +1,112 @@
 import * as React from 'react';
 
-import { Button, HeaderLayout } from '@strapi/design-system';
-import { CheckPermissions } from '@strapi/helper-plugin';
+import {
+  Box,
+  Button,
+  ContentLayout,
+  Flex,
+  HeaderLayout,
+  Main,
+  Tab,
+  TabGroup,
+  TabPanel,
+  TabPanels,
+  Tabs,
+} from '@strapi/design-system';
+import {
+  AnErrorOccurred,
+  CheckPermissions,
+  LoadingIndicatorPage,
+  PageSizeURLQuery,
+  PaginationURLQuery,
+  useQueryParams,
+} from '@strapi/helper-plugin';
 import { Plus } from '@strapi/icons';
 import { useIntl } from 'react-intl';
 
 import { AddReleaseDialog } from '../components/AddReleaseDialog';
+import { ReleasesGrid } from '../components/ReleasesGrid';
 import { PERMISSIONS } from '../constants';
+import { useGetReleasesQuery, GetAllReleasesQueryParams } from '../services/release';
 
 const ReleasesPage = () => {
   const [addReleaseDialogIsShown, setAddReleaseDialogIsShown] = React.useState(false);
   const { formatMessage } = useIntl();
+  const [{ query }, setQuery] = useQueryParams<GetAllReleasesQueryParams>();
+  const response = useGetReleasesQuery(query);
 
-  const total = 0; // TODO: replace it with the total number of releases
+  const { isLoading, isSuccess, isError } = response;
 
   const toggleAddReleaseDialog = () => {
     setAddReleaseDialogIsShown((prev) => !prev);
   };
 
+  if (isLoading) {
+    return (
+      <Main aria-busy={isLoading}>
+        <HeaderLayout
+          title={formatMessage({
+            id: 'content-releases.pages.Releases.title',
+            defaultMessage: 'Releases',
+          })}
+          primaryAction={
+            <CheckPermissions permissions={PERMISSIONS.create}>
+              <Button startIcon={<Plus />} onClick={toggleAddReleaseDialog}>
+                {formatMessage({
+                  id: 'content-releases.header.actions.add-release',
+                  defaultMessage: 'New release',
+                })}
+              </Button>
+            </CheckPermissions>
+          }
+        />
+        <ContentLayout>
+          <LoadingIndicatorPage />
+        </ContentLayout>
+      </Main>
```

Comment:
Maybe this could be a composed component outside of page component? So we don't have to repeat it below...?
```tsx
const ReleaseLayout = ({ children, isLoading }) => {
	return <Main aria-busy={isLoading}>
        <HeaderLayout
          title={formatMessage({
            id: 'content-releases.pages.Releases.title',
            defaultMessage: 'Releases',
          })}
          primaryAction={
            <CheckPermissions permissions={PERMISSIONS.create}>
              <Button startIcon={<Plus />} onClick={toggleAddReleaseDialog}>
                {formatMessage({
                  id: 'content-releases.header.actions.add-release',
                  defaultMessage: 'New release',
                })}
              </Button>
            </CheckPermissions>
          }
        />
        <ContentLayout>
					{children}
        </ContentLayout>
      </Main>
}
```

Reply:
tbh the wrapper outside the ContentLayout are different if you are on the onloading state or not, but yeah I can try to abstract it a little bit

Reply:
If it doesn't make sense that's fine, I thought they were very similar so thought maybe it could work